packages/gatsby-theme-aio/algolia/index-records.js

```diff
@@ -49,6 +49,10 @@ function indexRecords(isDryRun) {
 
         const markdownFiles = [];
         for (const node of nodes) {
+
+          // skip file if noIndex is set true in frontmatter
+          if (node.childMdx.frontmatter.noIndex) continue;
```

Comment:
Hmm I think we might to do something like:
`if (node?.childMdx?.frontmatter?.noIndex) continue;`

To avoid any undefined errors if users don't have a front matter definition. Not sure if `node` has an implicit `node.childMdx.frontmatter` children. If so, don't need to add the optional chaining.

Reply:
I had the same thought originally but then realized we handle fallbacks for the MdxFrontmatter with createResolver inside `gatsby-node.js`. I have it set to default `false` and it passed my testing.